.github/workflows/ci.yml

```diff
@@ -46,6 +46,9 @@ jobs:
         with:
           distribution: 'zulu'
           java-version: ${{ matrix.java-version }}
+      - name: Configure Problem Matchers
+        run: |
+          echo "::add-matcher::.github/problem-matcher.json"
```

Comment:
If we have annotate.yml  as a separate flow, is this still needed?
are the two related?

Reply:
No, those two are not related. One is for Maven check plugins and the Error Prone compiler, the separate workflow is to process test results (only). I probably should have created two separate PRs but, as usual, I thought it'll be a simpler change.

Reply:
Maybe rename annotate.yml to `annotate-surefire-results`?

Also is the name of the problem matcher file required to be `problem-matcher.json`? If not a helpful name would be useful (since the regex doesn't make it obvious which problems it's matching).

Reply:
This workflow annotates both Surefire and TestNG reports. If we want to use `annotate` as a verb, it actually annotates a (new) check run using test reports as the source, so `annotate-using-test-reports` but it starts to get too verbose.

There's no requirement for the Problem Matcher file name as it is referenced by the name. It can contain different sets of rules. It would make sense to use a less generic name if there were different sets used in different jobs, but we only have one now. The `owner` field inside the file should somehow explain what's the rule for. I put `maven` there but I'm open to suggestions.

Reply:
owner = maven_errorprone_checkstyle works for me. I don't care about verbosity as long as I don't require reverse engineering the regex.

For the `annotate` I think existing is fine too (what else would be annotating other than test results?) since we annotate both normal and product tests.

Reply:
Done!